src/main/java/org/assertj/core/error/ShouldNotBeEqualComparingFieldByFieldRecursively.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import org.assertj.core.internal.ComparisonStrategy;
+import org.assertj.core.internal.StandardComparisonStrategy;
+
+public class ShouldNotBeEqualComparingFieldByFieldRecursively extends BasicErrorMessageFactory {
+  /**
+   * Creates a new <code>{@link BasicErrorMessageFactory}</code>.
+   *
+   * @param format    the format string.
+   * @param arguments arguments referenced by the format specifiers in the format string.
+   */
+  public ShouldNotBeEqualComparingFieldByFieldRecursively(String format, Object... arguments) {
+    super(format, arguments);
+  }
+
```

Comment:
add `shouldNotBeEqualComparingFieldByFieldRecursively` static method to build the message
make this constructor private
add a test for it (see ShouldHaveBinaryContent_create_Test for a good example)